### PR TITLE
feat: support updating web hook url [MD-482]

### PR DESCRIPTION
**Pull request by jgongd**
## Ticket

MD-482

## Description

Users can edit the url of a webhook.
image

## Test Plan

CI passes.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/webhooks/postgres_webhook.go [hunk 1]
```diff
@@ -723,3 +724,25 @@ WHERE q.id = webhook_events_queue.id RETURNING webhook_events_queue.*
 	}
 	return &eventBatch{tx: &tx, events: events}, nil
 }
+
+// updateWebhook updates a webhook in the database.
+func (l *WebhookManager) updateWebhook(
```
Comment (gt2345) [inline master/internal/webhooks/postgres_webhook.go:729]: `updateWebhook` seems not related to `WebhookManager`, maybe we can make it a standalone function?

File: proto/src/determined/api/v1/webhook.proto [hunk 2]
```diff
@@ -86,3 +86,25 @@ message TestWebhookResponse {
...
+}
+
+// Response to PatchWebhookRequest.
+message PatchWebhookResponse {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "webhook" ] }
+  };
+
+  // The updated webhook.
+  determined.webhook.v1.PatchWebhook webhook = 1;
```
Comment (gt2345) [inline proto/src/determined/api/v1/webhook.proto:109]: nit: I'd prefer reusing `determined.webhook.v1.Webhook`

File: webui/react/src/components/WebhookEditModal.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        text: 'Save Changes',
+      }}
+      title="Edit Webhook">
+      <Form
+        autoComplete="off"
+        form={form}
+        id={idPrefix + FORM_ID}
+        layout="vertical"
+        onFieldsChange={onChange}>
+        <Form.Item
```
Comment (gt2345) [inline webui/react/src/components/WebhookEditModal.tsx:72]: Can we please pre-fill the old URL, and also display the name/id of the webhook in the modal

**Review by gt2345 — COMMENTED** [review]
Looks good! Just some minor comments.

File: master/internal/webhooks/postgres_webhook.go (hunk 1 above)
Comment (jgongd) [reply master/internal/webhooks/postgres_webhook.go:749]: Yeah, it wasn't related to `WebhookManager`. I ended up adding a feature that updates url of webhook in the cache. `if err := l.editTriggers(ts); err != nil {`

**Review by gt2345 — APPROVED** [review]
LGTM

File: master/internal/webhooks/postgres_webhook.go [hunk 4]
```diff
@@ -122,6 +123,26 @@ func (l *WebhookManager) removeTriggers(triggers []*Trigger) error {
 	return nil
 }
 
+func (l *WebhookManager) editTriggers(ts []*Trigger) error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	for _, t := range ts {
+		if t.TriggerType != TriggerTypeTaskLog {
+			return nil
```
Comment (salonig23) [inline master/internal/webhooks/postgres_webhook.go:132]: is it expected that you receive requests to edit non-TaskLog triggers? Also do we want to exit the function as soon as we encounter the first non-Task log type trigger?

Comment (jgongd) [reply master/internal/webhooks/postgres_webhook.go:132]: > is it expected that you receive requests to edit non-TaskLog triggers? 
Yes. I'm skipping it because only TaskLog triggers are stored in cache. https://github.com/determined-ai/determined/pull/9890/files/c08e8b902b779999dee41e3d7b287f67453d8750#diff-3ea86afd178c30910e3fdbc3f40b950a61d566b1e70bc4d932bbba3e693c2802R71-R78

>  Also do we want to exit the function as soon as we encounter the first non-Task log type trigger?
Good catch! I shouldn't exit the function.

**Review by salonig23 — APPROVED** [review]
LGTM!

**Comment by jgongd** [comment]
The failures in CI `test-intg-master` are known flakes.
